### PR TITLE
Ramdump support for Realtek Board

### PR DESCRIPTION
**Pull request by thapav**
This patch adds ramdump support fo rrealtek board.

### CODE REVIEW

**Comment by thapav** [comment]
Logs are as below:

## Option 1
```
Please enter serial port adapter:
For example: /dev/ttyUSB1 or /dev/ttyACM0
Enter:
/dev/ttyUSB0
Target Handshake successful do_handshake
Target entered to ramdump mode

=========================================================================
Ramdump Region Options:
1. ALL
2. Region 0:    ( Address: 0x10047418, Size: 216040 )    [Heap index = 0]
3. Region 1:    ( Address: 0x02000000, Size: 4194304 )   [Heap index = 0]
=========================================================================

Please enter desired ramdump option as below:
        1 for ALL
        2 for Region 0
        25 for Region 0 & 3 ...
Please enter your input: 2

No. of Regions to be dumped received: ramdump_recv

Receiving ramdump......

=========================================================================
Dumping Region: 0, Address: 0x10047418, Size: 216040bytes
=========================================================================
[=============>]
Ramdump received successfully..!
test@VirtualBox ~/tizenRTGH/tools/ramdump (download) 
```

##Option 2
```
Please enter serial port adapter:
For example: /dev/ttyUSB1 or /dev/ttyACM0
Enter:
/dev/ttyUSB0
Target Handshake successful do_handshake
Target entered to ramdump mode

=========================================================================
Ramdump Region Options:
1. ALL
2. Region 0:    ( Address: 0x10047418, Size: 216040 )    [Heap index = 0]
3. Region 1:    ( Address: 0x02000000, Size: 4194304 )   [Heap index = 0]
=========================================================================

Please enter desired ramdump option as below:
        1 for ALL
        2 for Region 0
        25 for Region 0 & 3 ...
Please enter your input: 3

No. of Regions to be dumped received: ramdump_recv

Receiving ramdump......

=========================================================================
Dumping Region: 1, Address: 0x02000000, Size: 4194304bytes
=========================================================================
[================================================================================================================================================================================================================================================================>]
Ramdump received successfully..!
test@VirtualBox ~/tizenRTGH/tools/ramdump (download)
```

File: os/arch/arm/src/amebad/chip.h [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+/************************************************************************************
+ * Public Data
+ ************************************************************************************/
+
+/************************************************************************************
+ * Public Functions
+ ************************************************************************************/
+
+#endif							/* __ARCH_ARM_INCLUDE_IMXRT_CHIP_H */
```
Comment (amandeepc) [inline os/arch/arm/src/amebad/chip.h:142]: Correction -> __ARCH_ARM_INCLUDE_AMEBAD_CHIP_H

File: os/arch/arm/src/amebad/chip.h [hunk 2]
```diff
@@ -0,0 +1,142 @@
+/****************************************************************************
```
Comment (sunghan-chang) [inline os/arch/arm/src/amebad/chip.h:1]: Could you let me know why this is necessry?
Can we remove including chip.h instead of adding this?

Comment (thapav) [reply os/arch/arm/src/amebad/chip.h:1]: Yes, that would be better.
I will remove it

File: os/arch/arm/src/amebad/chip.h (hunk 1 above)
Comment (thapav) [reply os/arch/arm/src/amebad/chip.h]: removed the file

File: os/board/common/crashdump.c [hunk 3]
```diff
@@ -26,7 +26,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <chip.h>
```
Comment (sunghan-chang) [inline os/board/common/crashdump.c:29]: Could you confirm this does not cause build break on artik, imxrt and mtk?

Comment (thapav) [reply os/board/common/crashdump.c:29]: Verified successful build for the following configurations after enabling ramdump support:

# imxrt1020-evk/loadable_elf_apps
# imxrt1050-evk/loadable_elf_apps
# mt7686_hdk/loadable_apps
# artik053/tc

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**